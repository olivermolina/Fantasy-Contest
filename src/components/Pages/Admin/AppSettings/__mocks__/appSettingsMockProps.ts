export const appSettingsMockProps = {
  appSettings: {
    BONUS_CREDIT_FREE_ENTRY_EQUIVALENT: '25',
    STAKE_TYPE_FREE_ENTRY: 'ALL_IN,INSURED',
    NUMBER_OF_PLAYERS_FREE_ENTRY: '2,3,4',
    MAX_MATCH_DEPOSIT_AMOUNT: '100',
    DEPOSIT_AMOUNT_OPTIONS: '10,25,50,75,100,200',
    RELOAD_BONUS_TYPE: 'FLAT',
    RELOAD_BONUS_AMOUNT: '0',
    MAX_RETENTION_BONUS: '50',
    RETENTION_BONUS_WEEKLY_CHANCE: '1',
    RETENTION_BONUS_MATCH_MULTIPLIER: '1',
    REFERRAL_CREDIT_AMOUNT: '25',
    REFERRAL_CUSTOM_TEXT:
      'Refer a friend from the link or code in your profile and get $25 in bonus credit!!',
    MIN_BET_AMOUNT: '1',
    MAX_BET_AMOUNT: '50',
    MAX_MARKET_ODDS: '150',
    MIN_MARKET_ODDS: '-150',
    initial: false,
    CHALLENGE_PROMO_MESSAGE:
      'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
    REPEAT_ENTRIES_LIMIT: '1',
  },
  onSubmit: () => console.log('onSubmit'),
};
