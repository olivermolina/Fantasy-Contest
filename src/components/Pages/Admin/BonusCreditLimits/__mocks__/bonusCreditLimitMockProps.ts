import { BetStakeType } from '@prisma/client';

export const bonusCreditLimitMockProps = {
  onSubmit: () => console.log('submit'),
  data: {
    signupFreeEntry: false,
    bonusCreditFreeEntryEquivalent: 50,
    bonusCreditLimits: [
      {
        contestCategoryId: '1',
        numberOfPicks: 2,
        id: '30387acd-660c-46f0-a9e6-af87c007e6fb',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 25,
      },
      {
        contestCategoryId: '2',
        numberOfPicks: 3,
        id: '316555e5-b447-468a-b57f-381bbf64919a',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 50,
      },
      {
        contestCategoryId: '3',
        numberOfPicks: 4,
        id: 'fc063fcd-1a25-46e2-9361-d947eec1f9a9',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 40,
      },
      {
        contestCategoryId: '4',
        numberOfPicks: 5,
        id: 'a2542740-e90f-4763-8a3a-f7afba68898c',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 20,
      },
    ],
  },
  openAddUsersFreeEntryConfirmDialog: () => console.log('open'),
};
