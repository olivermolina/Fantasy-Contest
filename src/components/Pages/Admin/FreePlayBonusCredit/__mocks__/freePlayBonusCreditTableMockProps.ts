import { PaymentStatusCode } from '~/constants/PaymentStatusCode';
import { TransactionType } from '@prisma/client';

export const freePlayBonusCreditTableMockProps = {
  onSubmitDeletePlayFreeCredit: () => console.log('delete'),
  data: [
    {
      id: '1',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.COMPLETE,
      type: TransactionType.CREDIT,
    },
    {
      id: '2',
      transactionDate: new Date(),
      amountBonus: 100,
      status: PaymentStatusCode.CANCELLED,
      type: TransactionType.DEBIT,
    },
  ],
  totalBonusCreditBalance: 0,
};
