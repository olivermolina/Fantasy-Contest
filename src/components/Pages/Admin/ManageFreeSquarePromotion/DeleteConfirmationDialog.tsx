import * as React from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';

interface Props {
  freeSquareId?: string;
  handleDelete: (freeSquareId: string) => void;
  open: boolean;
  setOpen: (open: boolean) => void;
}

export default function DeleteConfirmationDialog(props: Props) {
  const { handleDelete, open, setOpen, freeSquareId } = props;

  const closeDialog = () => {
    setOpen(false);
  };

  const confirmDelete = () => {
    closeDialog();
    if (freeSquareId) {
      handleDelete(freeSquareId);
    }
  };

  return (
    <Dialog
      open={open}
      onClose={closeDialog}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title">
        {'Delete Free Square Promotion?'}
      </DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description">
          Are you sure you want to remove free square promotion?
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={closeDialog}>Cancel</Button>
        <Button onClick={confirmDelete} variant={'outlined'} autoFocus>
          Confirm
        </Button>
      </DialogActions>
    </Dialog>
  );
}
