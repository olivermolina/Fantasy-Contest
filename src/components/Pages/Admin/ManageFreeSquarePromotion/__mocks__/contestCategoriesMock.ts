import { ContestCategory } from '@prisma/client';
import { faker } from '@faker-js/faker';

export const contestCategoriesMock: ContestCategory[] = [
  {
    id: faker.datatype.uuid(),
    numberOfPicks: 2,
    allInPayoutMultiplier: 3,
    primaryInsuredPayoutMultiplier: 2,
    secondaryInsuredPayoutMultiplier: 0.5,
  },
  {
    id: faker.datatype.uuid(),
    numberOfPicks: 3,
    allInPayoutMultiplier: 5,
    primaryInsuredPayoutMultiplier: 2.5,
    secondaryInsuredPayoutMultiplier: 1.25,
  },
  {
    id: faker.datatype.uuid(),
    numberOfPicks: 4,
    allInPayoutMultiplier: 10,
    primaryInsuredPayoutMultiplier: 5,
    secondaryInsuredPayoutMultiplier: 1.25,
  },
];
