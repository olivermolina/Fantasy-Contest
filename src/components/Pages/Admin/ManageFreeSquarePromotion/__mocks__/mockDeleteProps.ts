export const mockDeleteProps = {
  open: true,
  setOpen: () => console.log('open'),
  freeSquareId: '123',
  handleDelete: () => console.log('delete'),
};
