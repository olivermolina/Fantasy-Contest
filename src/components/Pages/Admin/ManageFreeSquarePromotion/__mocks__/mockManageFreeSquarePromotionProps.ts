import { League } from '@prisma/client';
import { contestCategoriesMock } from './contestCategoriesMock';

export const mockManageFreeSquarePromotionProps = {
  data: [
    {
      id: '63182-83-5227',
      selId: 5227,
      statName: 'Points',
      total: 11.5,
      matchName: 'STAN @ ORE ST',
      playerName: 'Glenn Taylor Jr.',
      marketId: '63182-83-5227',
      league: 'NCAAB',
      matchTime: 'Mar 02, 2023 @ 09:00 PM',
      freeSquare: {
        id: '1',
        discount: 95,
        pickCategories: [
          {
            id: '2 pick',
            numberOfPicks: 2,
          },
          {
            id: '3 pick',
            numberOfPicks: 3,
          },
          {
            id: '4 pick',
            numberOfPicks: 4,
          },
        ],
        maxStake: 25,
        freeEntryEnabled: false,
      },
      freeSquareDiscount: 95,
      discountedTotal: 1,
      picksCategory: '2,3,4',
      maxStake: 25,
      freeEntryEnabled: false,
    },
    {
      id: '63194-86-3541',
      selId: 3541,
      statName: 'Points Rebounds and Assists',
      total: 16.5,
      matchName: 'ARIZ @ USC',
      playerName: 'Kerr Kriisa',
      league: 'NCAAB',
      matchTime: 'Mar 02, 2023 @ 11:00 PM',
      freeSquare: {
        id: '1',
        discount: 98,
        pickCategories: [
          {
            id: '2 pick',
            numberOfPicks: 2,
          },
          {
            id: '3 pick',
            numberOfPicks: 3,
          },
          {
            id: '4 pick',
            numberOfPicks: 4,
          },
        ],
        maxStake: 25,
        freeEntryEnabled: false,
      },
      freeSquareDiscount: 95,
      discountedTotal: 1,
      picksCategory: '2,3,4',
      maxStake: 25,
      freeEntryEnabled: false,
    },
  ],
  setSelectedLeague: () => alert('setSelectedLeague'),
  handleSave: () => console.log('save'),
  contestCategories: contestCategoriesMock,
  selectedLeague: League.NBA,
  handleDelete: () => console.log('delete'),
};
