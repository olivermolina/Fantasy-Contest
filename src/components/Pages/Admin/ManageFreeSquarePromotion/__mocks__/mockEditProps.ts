import { mockManageFreeSquarePromotionProps } from './mockManageFreeSquarePromotionProps';
import { contestCategoriesMock } from './contestCategoriesMock';

export const mockEditProps = {
  row: mockManageFreeSquarePromotionProps.data[0],
  open: true,
  handleClose: () => console.log('close'),
  contestCategories: contestCategoriesMock,
  handleSave: () => console.log('save'),
};
