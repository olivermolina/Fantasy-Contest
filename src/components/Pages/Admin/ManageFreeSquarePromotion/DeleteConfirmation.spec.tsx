import { render } from '@testing-library/react';
import DeleteConfirmationDialog from './DeleteConfirmationDialog';
import { mockDeleteProps } from './__mocks__/mockDeleteProps';

describe('DeleteConfirmationDialog', () => {
  it('should render correctly', () => {
    const { container } = render(
      <DeleteConfirmationDialog {...mockDeleteProps} />,
    );
    expect(container).toMatchSnapshot();
  });
});
