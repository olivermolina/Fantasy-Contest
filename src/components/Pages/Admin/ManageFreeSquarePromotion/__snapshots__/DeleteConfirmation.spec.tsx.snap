// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteConfirmationDialog should render correctly 1`] = `
<div
  aria-hidden="true"
/>
`;
